### PR TITLE
Removed Git Basics section from contributing.md

### PR DESCRIPTION
**Pull request by aosdab**
Referencing issue [#45](https://github.com/exercism/docs/issues/45)

This time I left the heading up and referred to the documentation folder. If/when the git basics file gets merged, I can edit the link to refer to the actual file. Also, there's a little section at the bottom, I wasn't sure what to do with that...

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -707,265 +707,7 @@ Verify the config by calling `bin/configlet .` (notice the dot). This says
...
-```bash
-reword 433a0e1 added spinning wheel tests
-fixup 1f7d4aa pass spinning wheel
-fixup cf21737 oops
-fixup be4c410 rename example file
-fixup bb89a77 update config
-```
-
-[Further Reading](https://www.atlassian.com/git/tutorials/merging-vs-rebasing)
+Refer to [documentation](https://github.com/exercism/docs/tree/master/contributing-to-language-tracks).
```
Comment (Insti) [inline CONTRIBUTING.md]: * "documentation" is a bit vague for link text.
* Should the url point to git basics?

**Comment by Insti** [comment]
>  there's a little section at the bottom, I wasn't sure what to do with that...

Leaving it alone in this PR is probably the right thing to do, it's not related to Git basics.

**Comment by Insti** [comment]
It's useful if you link to the new location of the text you are removing in the pull request description. 
(Even if the location does not yet exist because the other PR has not been merged.)

**Comment by Insti** [comment]
Everything else looks good. 👍 :heart:

**Comment by aosdab** [comment]
Hi. I may have ran into some issues... I tried to edit the previous commit using `git commit --amend`  before I pulled the last changes. I got a bunch of conflicts! I hope I resolved them correctly. If you could help me out I would really appreciate it. Thanks!

**Comment by petertseng** [comment]
We are getting a little meta here (Git help in the PR that removes Git help)!!!

```
$ git log --graph master..pr819 --oneline                                                                                                                              (06-02 22:56)
* 65b4cd2 (HEAD -> pr819) Merge branch 'master' into docs-rm-git-basics
*   eaade38 Resolving merge conflict
|\
| * 95b77b1 Removed Git Basics section from CONTRIBUTING.md
* c027bdd Removed Git Basics section from CONTRIBUTING.md
```

There are two copies of the "Removed Git Basics section from CONTRIBUTING.md" commit in this PR which do different things (one before Insti's feedback and one after), and the two copies got merged together, and conflict as you would expect. It should instead be the case that there is only one copy.

So the conflict was resolved "correctly", but shouldn't be there in the first place. I recommend only including the changes from c027bdd. The various ways I can think of to achieve that are:

* `git rebase -i`, delete all lines except for c027bdd
* `git checkout docs-rm-git-basics; git reset master; git add CONTRIBUTING.md; git commit` (and enter the commit message). 
* substitute your preferred way of achieving the goal if you have one
* do nothing, maintainers **please** use *Squash and Merge* button when merging this PR

**Comment by petertseng** [comment]
> So the conflict was resolved "correctly"

I should remove scare quotes around "correctly". It was resolved correctly, no question about it, since the final product is the desired state.

**Comment by petertseng** [comment]
> `git rebase -i`

fine, I suppose I should say that should be `git rebase -i master` instead, otherwise it tries against the upstream tracking, which should obviously have no change

**Comment by petertseng** [comment]
Honestly I would consider `git reset --hard c027bdd` a reasonable way of doing this too, since none of the other commits matter.

**Comment by aosdab** [comment]
@petertseng Thank you for that wonderfully detailed explanation! As you can see I'm still a git newbie. :-) That `git log --graph` command is pretty cool. This has been a valuable experience and I appreciate being given the chance by @kytrinyx and @Insti. I will make whatever changes they see fit

**Comment by kytrinyx** [comment]
@aosdab How about you pick one of @petertseng's suggested approaches to fixing it—not the _do nothing_ :-)—and see if you can get it sorted. It's totally worth the git practice!

You'll need to `git push -f` up to your branch to update the pull request at the end of it.

Good luck ✨ 💓 ✨

**Comment by kytrinyx** [comment]
Oh, I forgot to say that no matter what you do, we can help recover if it goes sideways.

**Comment by aosdab** [comment]
Should be good to go. Thanks @kytrinyx :-)

**Comment by kytrinyx** [comment]
Usually I would fix this myself, but since we've been going meta on this one, here's one more thing that I'd like to suggest.

Notice how there are two commits, one is the change, and one is a merge commit from master?

Ideally, the PR would have just the first commit. The way to achieve that is:

- git checkout master
- git pull upstream master (this assumes that the exercism/x-common remote is named 'upstream' and that your own fork is called 'origin')
- git checkout docs-rm-git-basics
- git rebase master
- git push origin docs-rm-git-basics

This will put the commits from master _before_ your commit in this branch, thereby eliminating the merge commit.

Does that make sense?

**Comment by aosdab** [comment]
Yeah that makes sense. Thank you! So did this happen because I did not pull from master first before pushing? I was worried that if I pulled from master I would lose all the changes on this branch.

**Comment by petertseng** [comment]
> But since we're merging branches into master, it doesn't seem to make sense to merge master into other branches.

(I do think that they would produce the same result in the files, so the difference is just what the history looks like)

**Comment by kytrinyx** [comment]
> the difference is just what the history looks like

Yepp—that's the only difference. Some people don't care about the history, and some people do. There's no right or wrong, really, it's more about how the project works and the culture, and how people tend to need to use the history. On Exercism we care quite a lot about the history, because so many people are involved, and we're always trying to hunt down how some decision was made, or the history of some change. Getting rid of the extraneous merge commits from master to a branch keeps things a teensy bit more readable.

**Comment by kytrinyx** [comment]
💛 🌻 ✨ 🌻 💛 **thank you** 💛 🌻 ✨ 🌻 💛

**Comment by aosdab** [comment]
Hey @kytrinyx I appreciate the work you put into keeping this tidy. It's taught me not to be sloppy with my commits... and work :p I learned a lot doing this

**Pull request merged**